core/api-server/redis/redis.go

```diff
@@ -23,22 +23,62 @@
 package redis
 
 import (
+	"context"
+	"strings"
+	"strconv"
+	"fmt"
 	"github.com/NethServer/ns8-scratchpad/core/api-server/configuration"
 
 	"github.com/go-redis/redis/v8"
 )
 
-var redisConnection *redis.Client
-
 func Instance() *redis.Client {
 	// init client
-	redisConnection = redis.NewClient(&redis.Options{
+	return redis.NewClient(&redis.Options{
 		Network:  "tcp",
 		Addr:     configuration.Config.RedisAddress,
 		Username: configuration.Config.RedisUser,
 		Password: configuration.Config.RedisPassword,
 		DB:       0, // redis database
+		OnConnect: setClientNameCallback,
 	})
+}
+
+/*
+ * Check the client with name "searchName" has an idle time less than "limit"
+ */
+func CheckClientIdle(ctx context.Context, rdb *redis.Client, searchName string, limit int) (bool, error) {
+	// Consider only "normal" clients: agents are among them
+	val, err := rdb.Do(ctx, "CLIENT", "LIST", "TYPE", "normal").Result()
+	if err != nil {
+		return false, fmt.Errorf("Redis CLIENT LIST command failed (%v)", err)
+	}
+	// Split the CLIENT LIST output by lines and spaces
```

Comment:
This is an example of the string we are parsing

    127.0.0.1:6379> CLIENT LIST TYPE normal

```
id=8 addr=127.0.0.1:45040 laddr=127.0.0.1:6379 fd=8 name=cluster age=7608 idle=3 flags=b db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=19 obl=0 oll=0 omem=0 tot-mem=20531 events=r cmd=brpop user=cluster redir=-1
id=101 addr=127.0.0.1:45304 laddr=127.0.0.1:6379 fd=34 name=module/ldapproxy1 age=6404 idle=5 flags=b db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=29 obl=0 oll=0 omem=0 tot-mem=20541 events=r cmd=brpop user=module/ldapproxy1 redir=-1
id=241 addr=127.0.0.1:45806 laddr=127.0.0.1:6379 fd=23 name=module/samba1 age=6324 idle=4 flags=b db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=25 obl=0 oll=0 omem=0 tot-mem=20537 events=r cmd=brpop user=module/samba1 redir=-1
id=95 addr=127.0.0.1:45292 laddr=127.0.0.1:6379 fd=32 name=module/loki1 age=6405 idle=2 flags=b db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=24 obl=0 oll=0 omem=0 tot-mem=20536 events=r cmd=brpop user=module/loki1 redir=-1
id=9 addr=127.0.0.1:45042 laddr=127.0.0.1:6379 fd=9 name=node/1 age=7608 idle=2 flags=b db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=18 obl=0 oll=0 omem=0 tot-mem=20530 events=r cmd=brpop user=node/1 redir=-1
id=51 addr=127.0.0.1:45140 laddr=127.0.0.1:6379 fd=18 name=module/traefik1 age=7603 idle=4 flags=b db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=27 obl=0 oll=0 omem=0 tot-mem=20539 events=r cmd=brpop user=module/traefik1 redir=-1
id=142 addr=127.0.0.1:45532 laddr=127.0.0.1:6379 fd=27 name=module/promtail1 age=6376 idle=3 flags=b db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=28 obl=0 oll=0 omem=0 tot-mem=20540 events=r cmd=brpop user=module/promtail1 redir=-1
id=383 addr=127.0.0.1:46490 laddr=127.0.0.1:6379 fd=12 name= age=5 idle=1 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=48 qbuf-free=40906 argv-mem=20 obl=0 oll=0 omem=0 tot-mem=61492 events=r cmd=client user=cluster redir=-1
```